packages/cephalon/src/actions/leave-voice.scope.ts

```
@@ -2,40 +1,44 @@
-        setVoiceStateHandler: (h: VoiceStateChangeHandler) => {
-          ctx.bot!.voiceStateHandler = h;
-        },
-      })
-    : { leaveGuild: async () => false };
+	// If we have a bot, create a Discord-specific adapter. Otherwise, make a no-op adapter.
+	const voice = ctx?.bot
+		? makeDiscordVoiceAdapter({
+				client: ctx.bot.client,
+				getCurrentVoiceSession: () => ctx.bot!.currentVoiceSession,
+				setCurrentVoiceSession: (v: any) => {
+					ctx.bot!.currentVoiceSession = v;
```

ESLint

[error] Modifying an existing object/array is not allowed.

(functional/immutable-data)